Nodejs/Product/Npm/SPI/RootPackage.cs

```diff
@@ -47,6 +48,20 @@ internal class RootPackage : IRootPackage
             {
                 // otherwise we fail to create it completely...
             }
+
+            if (this.PackageJson != null)
+            {
+                this.Name = this.PackageJson.Name;
+            }
+            else
+            {
+                // this is the root package so the full folder name after node_nodules is the name
+                // of the package
+                var index = this.Path.IndexOf(NodejsConstants.NodeModulesFolder, StringComparison.OrdinalIgnoreCase);
```

Comment:
Does it need to be surrounded by path separators?  What if there's a "node_modules2" folder?

Reply:
I believe we always construct the path to contain 'node_modules' but wrapping in path separators should cover that.